plotting/plot_window/base_plot_window.py

```diff
@@ -17,11 +17,9 @@ def __init__(self,number,manager):
         self._import_aliases = {'plotting.pyplot': 'plt'} # the aliases used in script generation
 
     def set_as_kept(self):
-        self.canvas.axes.hold(False)
         self._display_status("kept")
 
     def set_as_current(self):
-        self.canvas.axes.hold(True)
         self._display_status("current")
 
     def display_status(self,status):
```

Comment:
Issue #25 was the result of make currenting enabling overplotting,
This issue in no longer present after migrating it to imshow as imshow clears the axes regardless of its state. But the last commit in this branch resolves the issue from the root